### PR TITLE
prov/gni: gnitest compiles with --enable-direct=gni.

### LINKED ISSUE #754: Compile gnitest with FABRIC_DIRECT when configured with --enable-direct=gni

### PR DESCRIPTION
**Pull request by e-harvey**
- Added new header file to conditionally define FABRIC_DIRECT and include "rdma/*.h"
- Removed "rdma/*.h" includes from gnitest source files and included the new header file.

Fixes #754 
@hppritcha

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by e-harvey** [comment]
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by e-harvey** [comment]
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by hppritcha** [comment]
hmm... i think jenkins is going to say confused about this commit.

File: prov/gni/include/gnix_gnitest_fi_direct.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2015-2016 Los Alamos National Security, LLC.
+ *                         All rights reserved.
```
Comment (hppritcha) [inline prov/gni/include/gnix_gnitest_fi_direct.h]: just use 2016 here since the file didn't exist in 2015.

**Comment by hppritcha** [comment]
:+1: modulo the little copyright change

**Comment by sungeunchoi** [comment]
The new header file should be put in prov/gni/test, and not in the provider include directory.  Also, it's needed generally and not just for FI_DIRECT, so it should be renamed to something more generic like `gnix_headers.h`.

**Comment by sungeunchoi** [comment]
Also, use `#include "gnix_headers.h"` rather than the brackets.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by e-harvey** [comment]
Currently, the compiler cannot find the direct headers since the top-level libfabric include/rdma directory doesn't contain them, however no warnings or errors are generated.  Instead, the compiler just uses the non-direct headers.  Modify gni_CPPFLAGS to include libfabric/prov/gni/include so that the direct headers are used when compiling gnitest with --enable-direct=gni.

**Comment by sungeunchoi** [comment]
In talking with Evan, I think this is sort of an artificial problem caused by the fact that we're not using an actual installed version of the header files.  I've opened #771 to set up the infrastructure to provide the gni include path to gcc when compiling gnitest.

**Comment by e-harvey** [comment]
bot:retest

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by e-harvey** [comment]
The jenkins test is failing because jenkins cannot find the correct sha for this PR.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by e-harvey** [comment]
Closing to squash commits and open a new PR.

**Pull request closed without merging**